src/pages/search/search.tsx

```diff
@@ -1,82 +1,76 @@
-import { Component } from 'react';
+import { type FC, useCallback, useEffect, useState } from 'react';
 import style from './search.module.scss';
 import SearchForm from '../../components/search-form/search-form';
-import CardList from '../../components/card-list/card-list';
 import type { Character } from '../../core/interfaces/interface.ts';
-import ErrorBoundary from '../../components/error-boundary/error-boundary';
 import { getCharacters } from '../../core/services/api-service.ts';
-import bgPath from '../../assets/image/rick-and-morty-bg.jpg';
 import Loading from '../../components/loading/loading.tsx';
 import Button from '../../components/ui/button/button.tsx';
+import { useLocalStorage } from '../../core/hooks/useLocalStorage.ts';
+import { LOCAL_STORAGE_KEY } from '../../core/constants/constants.ts';
+import { usePagination } from '../../core/hooks/usePagination.ts';
+import { ResultLayout } from '../../components/result-layout/ResultLayout.tsx';
+import { useQueryUpdate } from '../../core/hooks/useQuery.ts';
+import { SearchControls } from '../../components/search-controls/search-controls.tsx';
 
-interface ISearchState {
-  charList: Character[] | undefined;
-  query: string;
-  loading: boolean;
-}
+const Search: FC = () => {
+  const [charList, setCharList] = useState<Character[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [showControls, setShowControls] = useState(false);
+  const { savedQuery, setQueryLS } = useLocalStorage(LOCAL_STORAGE_KEY);
+  const { page, resetPage, maxPage, setMaxPage, prevPage, nextPage } =
+    usePagination();
+  useQueryUpdate(page, savedQuery);
 
-class Search extends Component {
-  state: ISearchState = {
-    query: '',
-    charList: undefined,
-    loading: false,
-  };
+  const handleClick = useCallback(
+    (name: string, queryPage: number) => {
+      setShowControls(false);
```

Comment:
```suggestion
      if (name !== savedQuery) {
        setQueryLS(name);
      }

      setShowControls(false);
```

you can add these lines here and there will not be a need to pass additional callback into SearchForm

Reply:
I get an endless loop of rerendering.